### PR TITLE
HBASE-28532 Remove vulnerable dependencies: slf4j-log4j12 and log4j:log4j

### PR DESCRIPTION
**Pull request by nikita15p**
Removed all deps of log4j 1.x which is vulnerable
Build and tests are passing 
```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Apache HBase Operator Tools 1.3.0-SNAPSHOT:
[INFO] 
[INFO] Apache HBase Operator Tools ........................ SUCCESS [  2.353 s]
[INFO] Apache HBase - Table Reporter ...................... SUCCESS [ 12.604 s]
[INFO] Apache HBase - HBCK2 ............................... SUCCESS [ 13.915 s]
[INFO] Apache HBase - HBase Tools ......................... SUCCESS [  5.482 s]
[INFO] Apache HBase Operator Tools - Assembly ............. SUCCESS [  0.188 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  34.954 s
[INFO] Finished at: 2024-08-08T20:18:15Z
[INFO] ------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 32s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   0m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m 19s |  master passed  |
| +1 :green_heart: |  spotless  |   0m  8s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  7s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 21s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  2s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  spotless  |   0m  6s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  javadoc  |   0m 12s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m  8s |  hbase-table-reporter in the patch passed.  |
| +1 :green_heart: |  unit  |   6m 48s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   |  12m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Operator-Tools-PreCommit/job/PR-142/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-operator-tools/pull/142 |
| Optional Tests | dupname asflicense javac javadoc unit spotless xml compile |
| uname | Linux 863c0a17ac48 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / 4286235 |
| Default Java | Oracle Corporation-1.8.0_342-b07 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Operator-Tools-PreCommit/job/PR-142/1/testReport/ |
| Max. process+thread count | 1256 (vs. ulimit of 5000) |
| modules | C: hbase-table-reporter . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Operator-Tools-PreCommit/job/PR-142/1/console |
| versions | git=2.30.2 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by guluo2016** [comment]
Executing `hbase hbck -j base-hbck2-1.3.0-SNAPSHOT.jar --help` sucessfully after applying this PR.

```shell
./bin/hbase hbck -j /home/software/hbase-operator-tools-master/hbase-hbck2/target/hbase-hbck2-1.3.0-SNAPSHOT.jar --help
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/software/hbase-operator-tools-master/hbase-hbck2/target/hbase-hbck2-1.3.0-SNAPSHOT.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/test/hbase-4.0.0-alpha-1-SNAPSHOT/lib/client-facing-thirdparty/log4j-slf4j-impl-2.17.2.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
usage: HBCK2 [OPTIONS] COMMAND <ARGS>
Options:
 -d,--debug                                       run with debug output
 -h,--help                                        output this help message
 -p,--hbase.zookeeper.property.clientPort <arg>   port of hbase ensemble
 -q,--hbase.zookeeper.quorum <arg>                hbase ensemble
 -s,--skip                                        skip hbase version check
                                                  (PleaseHoldException)
 -v,--version                                     this hbck2 version
 -z,--zookeeper.znode.parent <arg>                parent znode of hbase
                                                  ensemble
```

**Comment by guluo2016** [comment]
Hi @Apache9 ,  Could you take a look? 
I found that [LOG4J2-3341](https://issues.apache.org/jira/browse/LOG4J2-3341) was created by you.

 Details see: [HBASE-28531](https://issues.apache.org/jira/browse/HBASE-28531) and [HBASE-28532](https://issues.apache.org/jira/browse/HBASE-28532)

**Review by guluo2016 — APPROVED** [review]

File: hbase-table-reporter/pom.xml [hunk 1]
```diff
@@ -36,17 +36,23 @@
...
-      <artifactId>slf4j-log4j12</artifactId>
-      <version>1.7.25</version>
+      <groupId>org.apache.logging.log4j</groupId>
+      <artifactId>log4j-slf4j-impl</artifactId>
+      <version>${log4j2.version}</version>
     </dependency>
     <dependency>
       <groupId>org.apache.hbase</groupId>
       <artifactId>hbase-shaded-client</artifactId>
       <version>2.1.1</version>
```
Comment (Apache9) [inline hbase-table-reporter/pom.xml:49]: Why here we use 2.1.1, instead of the hbase.version declared in the parent pom?

Comment (guluo2016) [reply hbase-table-reporter/pom.xml:49]: Yes, `${hbase.version}` should be used in here.
Seems that didn't pay attention to these, This issue only focused on log4j dependency, perhaps also fix it in this PR  @nikita15p

Comment (Apache9) [reply hbase-table-reporter/pom.xml:49]: Can be fixed in another issue.

**Review by Apache9 — APPROVED** [review]

File: hbase-table-reporter/pom.xml (hunk 1 above)
Comment (NihalJain) [reply hbase-table-reporter/pom.xml:49]: Technically this dependecny should be not even needed post https://github.com/apache/hbase-operator-tools/pull/144 as I think log4j 1 would have been leaked via hbase 2.1.1. @nikita15p  could you please check and update

File: hbase-table-reporter/pom.xml [hunk 2]
```diff
@@ -36,17 +36,23 @@
...
+      <exclusions>
+        <exclusion>
```
Comment (NihalJain) [inline hbase-table-reporter/pom.xml:51]: this should no longer be needed with latest code

**Comment by NihalJain** [comment]
Hi @nikita15p could you please resolve conflicts here and update PR as needed.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 18s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 28s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   0m 30s |  master passed  |
| +1 :green_heart: |  compile  |   0m 37s |  master passed  |
| +1 :green_heart: |  spotless  |   0m  8s |  branch has no errors when running spotless:check.  |
| +1 :green_heart: |  javadoc  |   0m 22s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m  5s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   0m 55s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 38s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  3s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  spotless  |   0m  5s |  patch has no errors when running spotless:check.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m  6s |  hbase-table-reporter in the patch passed.  |
| +1 :green_heart: |  unit  |   4m 45s |  hbase-hbck2 in the patch passed.  |
| +1 :green_heart: |  unit  |   1m 32s |  hbase-tools in the patch passed.  |
| +1 :green_heart: |  unit  |   6m  7s |  root in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 19s |  The patch does not generate ASF License warnings.  |
|  |   |  18m 40s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Operator-Tools-PreCommit/job/PR-142/3/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-operator-tools/pull/142 |
| Optional Tests | dupname asflicense javac javadoc unit spotless xml compile |
| uname | Linux d005262a6bb8 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/jenkins/hbase-personality.sh |
| git revision | master / ff7dae1 |
| Default Java | Oracle Corporation-1.8.0_342-b07 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Operator-Tools-PreCommit/job/PR-142/3/testReport/ |
| Max. process+thread count | 1288 (vs. ulimit of 5000) |
| modules | C: hbase-table-reporter hbase-hbck2 hbase-tools . U: . |
| Console output | https://ci-hbase.apache.org/job/HBase-Operator-Tools-PreCommit/job/PR-142/3/console |
| versions | git=2.30.2 maven=3.8.6 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by NihalJain — APPROVED** [review]
nit: we should update title to capture that it also removes log4:log4j. Will handle that before merge.

**Comment by NihalJain** [comment]
Hi @guluo2016 I was looking to commit this PR. Could you provide your Name < Email ID > in format:  Nihal Jain <nihaljain@apache.org> so that I could add to commit message.

**Comment by guluo2016** [comment]
> Hi @guluo2016 I was looking to commit this PR. Could you provide your Name < Email ID > in following format so that I could add to commit message:
> 
> * `Nihal Jain <nihaljain@apache.org>`

I have updated my name.  If there are any issues, please let me know，thanks.

**Comment by NihalJain** [comment]
> > Hi @guluo2016 I was looking to commit this PR. Could you provide your Name < Email ID > in following format so that I could add to commit message:
> > 
> > * `Nihal Jain <nihaljain@apache.org>`
> 
> I have updated my name. If there are any issues, please let me know，thanks.

Thanks @guluo2016 for your reviews and testing. Added as Reviewed-by in commit message. Please see if this is right ```Peng Lu <lupeng_nwpu@qq.com>```

**Pull request merged**